konfera/models/schedule.py

```diff
@@ -23,3 +25,35 @@ class Schedule(KonferaModel):
 
     def __str__(self):
         return _('%(start)s (%(duration)s min)') % {'start': self.start, 'duration': self.duration}
+
+    def clean(self, *args, **kwargs):
+        # Only approved talk can be scheduled
+        if self.talk.status != 'approved':
+            raise ValidationError(_('You cannot schedule unapproved talks'))
+
+        # Event is related to location and location has room,
+        # make sure selected room in schedule belongs to Event's location
+        room = get_model('Room').objects.filter(location=event.location)
```

Comment:
@ZoltanOnody actually there is no super valid reason. i commonly use that instead of adding an import line inside the method. also i was avoiding circular importing